### PR TITLE
Add net module to cnex

### PR DESCRIPTION
**Pull request by gitlarryf**

### CODE REVIEW

File: exec/cnex/lib/socketx.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#else
+
+#include <arpa/inet.h>
+#include <netdb.h>
+#include <netinet/in.h>
+#include <sys/select.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+#define closesocket(x)      close(x);
```
Comment (ghewgill) [inline exec/cnex/lib/socketx.h]: Do not put a semicolon after a macro definition like this.

File: exec/cnex/array.c [hunk 2]
```diff
@@ -70,6 +70,18 @@ void array_freeArray(Array *self)
...
 
+void array_clearArray(Array *self)
+{
+    if (self != NULL && self->data != NULL) {
+        for (size_t i = 0; i < self->size; i++) {
+            cell_clearCell(&self->data[i]);
+        }
+        free(self->data);
+    }
+    self->data = NULL;
```
Comment (ghewgill) [inline exec/cnex/array.c]: Should this be inside the check for `self != NULL`? What if `self` actually is `NULL`?

File: exec/cnex/lib/net.c [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+        addr.s_addr = INADDR_ANY;
+    } else {
+        addr.s_addr = inet_addr(address);
+        if (addr.s_addr == INADDR_NONE) {
+            struct hostent *he = gethostbyname(address);
+            if (he == NULL) {
+                // ToDo: Raise exception here, that host does not exist?
+                goto bail;
+            }
+            addr = *(struct in_addr*)he->h_addr_list;
```
Comment (ghewgill) [inline exec/cnex/lib/net.c]: This probably won't work, you're not using `h_addr_list` correctly.

File: exec/cnex/lib/socketx.h (hunk 1 above)
Comment (gitlarryf) [reply exec/cnex/lib/socketx.h]: Fixed.

File: exec/cnex/array.c (hunk 2 above)
Comment (gitlarryf) [reply exec/cnex/array.c]: Good call!  Fixed.

File: exec/cnex/lib/net.c (hunk 3 above)
Comment (gitlarryf) [reply exec/cnex/lib/net.c]: Fixed.

**Pull request merged**